ftdetect/gentoo.vim

```diff
@@ -75,7 +75,7 @@ au BufNewFile,BufRead {*/thirdpartymirrors,*/portage/mirrors}
     \     set filetype=gentoo-mirrors
 
 " make.conf
-au BufNewFile,BufRead make.{conf,globals}
+au BufNewFile,BufRead {*/make.{conf,globals},*/portage/make.conf/*}
```

Comment:
I'm sorry but I don't know vimscript much. Could you explain to me why you had to add `*/` and it (apparently) wasn't necessary before?

Reply:
it means "any path before `portage/make.conf/`".
Among other things, it allows prefix paths to also match the pattern.

Reply:
Ok but previously `make.{conf,globals}` worked, so is `*/make.{conf,globals}` needed now or it's just… good practice?

Reply:
IIRC, not that it is needed (nor also breaks anything) for `make.{conf,globals}` as files.
But, (also, IIRC), it is needed for directory-contained paths.

I can't remember details, but I myself also had some problems when used paths without prepended `*/` in custom portage-related autocommands.

So, at least for `*/make.conf/*` I think it is needed.
And `/portage/` part, I guess, was used by PR author to limit match to only paths containing `portage` and don't match other `make.conf`-conaining paths.

// Btw, 

On the one hand, if we wouldn't need to keep backward comatibility with few-decades old systems, where `make.conf` was in just "${EROOT}/etc/`, and not in `${EROOT}/etc/portage`, I'd also suggest to add `/portage/` prefix to `make.{conf,globals}` too.

On the other hand, I'm using the layout with "shared" diectory (that syncs between mutiple devices) in `/etc/portage`, and following restrictions like `*/portage/*` makes things to not match :cry: .

So, I'm not sure which side to choose: the one about avoiding conflicts with random non-gentoo-related `make.conf`'s, or the one about suporting any possible paths :shrug:

Reply:
Ok, thanks.

That's a valid point but I don't think we really need to worry about random `make.conf` files matching right now. Let's cross that bridge when we get to it.

Reply:
I did it this way because that's what was there for other things in `/portage`. Personally I'm not concerned about non-gentoo related files matching these names, as they're pretty specific. Just wanted to keep up the consistency :)